src/utils/SpokeUtils.ts

```diff
@@ -307,12 +310,38 @@ export async function relayFillStatus(
 
   if (![FillStatus.Unfilled, FillStatus.RequestedSlowFill, FillStatus.Filled].includes(fillStatus)) {
     const { originChainId, depositId } = relayData;
-    throw new Error(`relayFillStatus: Unexpected fillStatus for ${originChainId} deposit ${depositId}`);
+    throw new Error(`relayFillStatus: Unexpected fillStatus for ${originChainId} deposit ${depositId} (${fillStatus})`);
   }
 
   return fillStatus;
 }
 
+export async function fillStatusArray(
+  spokePool: Contract,
+  relayData: V3RelayData[],
+  blockTag: BlockTag = "latest"
+): Promise<(FillStatus | undefined)[]> {
+  const fillStatuses = "fillStatuses";
+  const destinationChainId = await spokePool.chainId();
+  const queries = relayData.map((relayData) => {
+    const hash = getV3RelayHash(relayData, destinationChainId);
+    return spokePool.interface.encodeFunctionData(fillStatuses, [hash]);
+  });
+  const multicall = await spokePool.callStatic.multicall(queries, { blockTag });
+  const status = multicall.map(
+    (result: BytesLike) => spokePool.interface.decodeFunctionResult(fillStatuses, result)[0]
+  );
+
+  const bnUnfilled = toBN(FillStatus.Unfilled);
+  const bnFilled = toBN(FillStatus.Filled);
+
+  return status.map((status: unknown) => {
+    return BigNumber.isBigNumber(status) && status.gte(bnUnfilled) && status.lte(bnFilled)
+      ? status.toNumber()
+      : undefined;
```

Comment:
Can these ever be undefined? Wouldn't that indicate a contract error?

Reply:
The contract itself should never permit this to occur, but I don't think we can make that assumption of the RPC provider, so that's why the check is there. I don't expect that we'd ever see this arise in production, but the options would be to throw or return undefined and force the caller to check it and decide what they want to do.

Reply:
Also, the check is not exactly looking for whether the status returned is `undefined` - it's moreso checking that it fits within the range of values that we consider valid for this mapping.